boot/boot.go

```diff
@@ -426,10 +426,9 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
 
 	// write modeenv on the ubuntu-data partition
 	modeenv := &Modeenv{
-		Mode:           "run",
-		RecoverySystem: filepath.Base(bootWith.RecoverySystemDir),
```

Comment:
We actually need to know the recovery system to drive first boot seeding of the run system. We should have a TODO somewhere close to doMarkSeeed about maybe removing it at that point...

Reply:
Thanks! I readded this and added the missing TODO.